vsphere/resource_vsphere_entity_permissions.go

```diff
@@ -119,6 +122,11 @@ func resourceEntityPermissionsRead(d *schema.ResourceData, meta interface{}) err
 		permissionObj["role_id"] = strconv.Itoa(int(permission.RoleId))
 		permissionObjs = append(permissionObjs, permissionObj)
 	}
+
+	sort.Slice(permissionObjs[:], func(i, j int) bool {
```

Comment:
I noticed this pattern throughout, you are reslicing `slice[:]` and it's unclear to me the benefit (granted I'm not a slice ninja). This [go playground](https://play.golang.org/p/2BeLUXMWqts) demonstrates there is no difference. Curious on the reason.

Reply:
permissionObjs[:] It is converting array into slice.

Reply:
Pardon me, I thought that was needed to convert array into slice, but turns out permissionObjs is already a slice.